script.js

```diff
@@ -1,4 +1,246 @@
+    while (rankCounter <= 13) {
+      // By default, the card name is the same as rankCounter
+      var cardName = rankCounter;
+
+      // If rank is 1, 11, 12, or 13, set cardName to the ace or face card's name
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+      } else if (cardName == 12) {
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+
+      // Create a new card with the current name, suit, and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+
+      // Add the new card to the deck
+      cards.push(card);
+
+      // Increment rank Counter to iterate over the next rank
+      rankCounter += 1;
+    }
+
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+
+  // Return the completed card deck
+  return cards;
 };
+
+var makeDeckDone = makeDeck();
+
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cards array
+function shuffleDeck(cards) {
+  // Loop over the card deck array once
+  var index = 0;
+  while (index < cards.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cards.length);
+
+    // Select the card that corresponds to randomIndex
+    var randomItem = cards[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentItem = cards[index];
+    // Swap positions of randomCard and currentCard in the deck
+    cards[index] = randomItem;
+
+    cards[randomIndex] = currentItem;
+    // Increment currentIndex
+    index += 1;
+  }
+  // Return the shuffled deck
+  return cards;
+}
+
+// Function to Create NEW DECK
+
+function createNewDeck() {
+  var shuffleDeckDone = shuffleDeck(makeDeckDone);
+  return shuffleDeckDone;
+}
+
+//////// got blackjack?
+
+function gotBlackjack(handArray) {
+  var playerCardOne = handArray[0];
+  var playerCardTwo = handArray[1];
+
+  var gotBlackjack = false;
+
+  if (
+    (playerCardOne.name == "ace" && playerCardTwo.rank >= 10) ||
+    (playerCardOne.rank >= 10 && playerCardTwo.name == "ace")
+  ) {
+    gotBlackjack = true;
+  } else {
+    gotBlackjack = false;
+  }
+  return gotBlackjack;
+}
+
+function sumHandScore(handArray) {
+  var sum = 0;
+  var i = 0;
+
+  while (i < handArray.length) {
+    var currentCard = handArray[i];
+    if (
+      currentCard.name == "jack" ||
+      currentCard.name == "queen" ||
+      currentCard.name == "king"
+    ) {
+      sum = sum + 10;
+    } else {
+      sum = sum + currentCard.rank;
+    }
+    i += 1;
+  }
+  return sum;
+  ``;
+}
+
+// Game MODES
+var gameStart = "game start";
+var gameCardsIssued = "cards issued";
+var gameScore = "game score";
+var gameModeNow = gameStart;
+var playDeck = "";
```

Comment:
playDeck would ultimately be an array so a good default value should be an empty array. It is not a common practice for a variable to have different types i.e. from a string to an array